### PR TITLE
Add a person's name and date of birth to bookings

### PR DESCRIPTION
**Pull request by patrickjfl**
# Changes in this PR

This PR allows [the PoP's name and date of birth](https://trello.com/c/59XvRBTe/676-add-a-persons-name-and-dob-to-bookings) to be rendered on booking related pages.

To achieve this I have separated the locationDetailsHeader into two components, locationDetailsHeader and popDetailsHeader. The former header now exclusively renders location related data, while the latter header renders the CRN as before, together with the PoP's name and date of birth.

I have also added json representations of Persons available in the local and dev environments, to support end to end testing.

### Before

Screenshot 2023-03-21 at 13 50 24

Screenshot 2023-03-21 at 13 50 02

### After

Screenshot 2023-03-21 at 13 48 39

Screenshot 2023-03-21 at 13 49 35

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
    `production` in a backwards compatible way? (This includes our API,
    infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
    advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

### CODE REVIEW

**Comment by patrickjfl** [comment]
[Add a persons Name and DOB to bookings](https://trello.com/c/59XvRBTe/676-add-a-persons-name-and-dob-to-bookings)

File: server/controllers/temporary-accommodation/manage/arrivalsController.test.ts [hunk 1]
```diff
@@ -16,6 +17,11 @@ import ArrivalsController from './arrivalsController'
 
 jest.mock('../../../utils/validation')
 jest.mock('../../../utils/restUtils')
+jest.mock('../../../utils/bookingUtils', () => {
+  return {
+    bookingWithPersonDobInUiFormat: (booking: Booking) => booking,
+  }
+})
```
Comment (bryangaledxw) [inline server/controllers/temporary-accommodation/manage/arrivalsController.test.ts]: I think avoid this style of mock implementation if you can. Elsewhere, it's only used in awkward cases where we need to import a file where we want to mock some things but use the real versions of others

**Review by bryangaledxw — COMMENTED** [review]
Looks good, and wasn't quite as intimidating as I thought it would be from the changed files count! I would strongly suggest though, instead of using `bookingWithPersonDobInUiFormat`, instead call `formatDate` (see `nunjucksSetup.ts`) inside the pop-details-header macro

Then we keep the UI concerns self contained, and we don't have this call to a formatting method inside each controller

File: server/controllers/temporary-accommodation/manage/bookingsController.ts [hunk 2]
```diff
@@ -53,16 +54,30 @@ export default class BookingsController {
...
+          premises,
+          room,
+          person,
+          arrivalDate,
+          departureDate,
+        })
+      } catch (err) {
+        if (err.status === 404) {
+          insertGenericError(err, 'crn', 'doesNotExist')
+        }
```
Comment (bryangaledxw) [inline server/controllers/temporary-accommodation/manage/bookingsController.ts]: Should this also handle `403`, same as `create`?

**Review by bryangaledxw — APPROVED** [review]
Looks good, just want to check about handling 403s

File: server/controllers/temporary-accommodation/manage/bookingsController.ts (hunk 2 above)
Comment (patrickjfl) [reply server/controllers/temporary-accommodation/manage/bookingsController.ts]: oops yes, I forgot to add that. thanks!

**Pull request merged**